tests/PHPStan/Analyser/data/bug-10122.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Bug10122;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @param string $s
+ * @param numeric-string $ns
+ */
+function doFoo(string $s, $ns, float $f) {
+	assertType('string', ++$s);
```

Comment:
`++$s` will be an int even if the input value is a string. https://3v4l.org/i4rru

refs https://github.com/phpstan/phpstan/issues/10187

Reply:
please have another look, I think it should look better now